jmclient/jmclient/wallet_rpc.py

```diff
@@ -0,0 +1,653 @@
+            auth_json = self.get_POST_body(request, ["password"])
+            if not auth_json:
+                raise InvalidRequestFormat()
+            password = auth_json["password"]
+            if self.wallet_service is None:
+                wallet_path = get_wallet_path(walletname, None)
+                try:
+                    wallet = open_test_wallet_maybe(
+                            wallet_path, walletname, 4,
+                            password=password.encode("utf-8"),
+                            ask_for_password=False)
+                except StoragePasswordError:
+                    raise NotAuthorized("invalid password")
+                except RetryableStorageError as e:
+                    # .lock file exists
+                    raise NotAuthorized(repr(e))
+                except StorageError:
+                    # wallet is not there
+                    raise NoWalletFound()
+                return self.initialize_wallet_service(request, wallet, walletname)
+            else:
+                jlog.warn('Tried to unlock wallet, but one is already unlocked.')
+                jlog.warn('Currently only one active wallet at a time is supported.')
+                return make_jmwalletd_response(request,
+                                walletname=walletname,
+                                already_loaded=True)
+
+        #This route should return list of current wallets created.
+        @app.route('/wallet/all', methods=['GET'])
+        def listwallets(self, request):
+            wallet_dir = os.path.join(jm_single().datadir, 'wallets')
+            # TODO: we only allow .jmdat files, and assume they
+            # are actually wallets; but we should validate these
+            # wallet files before returning them (though JM itself
+            # never puts any other kind of file in this directory,
+            # the user conceivably might).
+            wallets = os.listdir(wallet_dir)
+            wallets = [w for w in wallets if w.endswith("jmdat")]
+            return make_jmwalletd_response(request, wallets=wallets)
+
+        #route to get external address for deposit
+        @app.route('/wallet/<string:walletname>/address/new/<string:mixdepth>',methods=['GET'])
+        def getaddress(self, request, walletname, mixdepth):
+            self.check_cookie(request)
+            if not self.wallet_service:
+                raise NoWalletFound()
+            mixdepth = int(mixdepth)
+            address = self.wallet_service.get_external_addr(mixdepth)
+            return make_jmwalletd_response(request, address=address)
+
+        def get_listutxos_response(self, utxos):
+            res = []
+            for k, v in utxos.items():
+                inner_dict = {}
+                inner_dict.update({"utxo": k})
+                for k2, v2 in v.items():
+                    inner_dict.update({k2: v2})
+                res.append(inner_dict)
+            return res
```

Comment:
Yeah that's functionally equivalent, will do.

Reply:
Ah I get your comment, this passed-by-ref object is not getting used somewhere else (I suppose the desire to avoid thinking about that is why I often default to creating new objects "by-hand". Well, not a huge deal but we may as well do the more performant/less code thing.